File: app/src/main/java/zac4j/com/webapp/WebAppActivity.java
package zac4j.com.webapp;

import android.annotation.SuppressLint;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.webkit.JsPromptResult;
import android.webkit.JsResult;
import android.webkit.WebChromeClient;
import android.webkit.WebView;

/**
 * Activity for web app.
 * Created by Zaccc on 2017/7/20.
 */

public class WebAppActivity extends AppCompatActivity {

  private WebView mBrowser;

  @SuppressLint("SetJavaScriptEnabled") @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    mBrowser = (WebView) findViewById(R.id.webview);
    mBrowser.getSettings().setJavaScriptEnabled(true);
    mBrowser.setWebChromeClient(new MyWebChromeClient());
    mBrowser.loadUrl("file:///android_asset/prompt.html");
  }

  private class MyWebChromeClient extends WebChromeClient {
    @Override
    public boolean onJsAlert(WebView view, String url, String message, final JsResult result) {
      new AlertDialog.Builder(WebAppActivity.this).setTitle("WebApp Alert")
          .setMessage(message)
          .setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
            @Override public void onClick(DialogInterface dialog, int which) {
              result.confirm();
            }
          })
          .setCancelable(false)
          .create()
          .show();
      return true;
    }

    @Override
    public boolean onJsConfirm(WebView view, String url, String message, final JsResult result) {
      new AlertDialog.Builder(WebAppActivity.this).setTitle("WebApp Confirm")
          .setMessage(message)
          .setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
            @Override public void onClick(DialogInterface dialog, int which) {
              result.confirm();
            }
          })
          .setCancelable(false)
          .create()
          .show();
      return true;
    }

    @Override
    public boolean onJsPrompt(WebView view, String url, String message, String defaultValue,
        final JsPromptResult result) {
      new AlertDialog.Builder(WebAppActivity.this).setTitle("WebApp Prompt")
          .setMessage(message)
          .setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
            @Override public void onClick(DialogInterface dialog, int which) {
              result.confirm();
            }
          })
          .setCancelable(false)
          .create()
          .show();
      return true;
    }
  }
}

File: app/src/main/java/zac4j/com/webapp/WebAppActivity.kt
package zac4j.com.webapp

import android.R.string
import android.annotation.SuppressLint
import android.os.Bundle
import android.view.View
import android.webkit.JsPromptResult
import android.webkit.JsResult
import android.webkit.WebChromeClient
import android.webkit.WebView
import androidx.appcompat.app.AlertDialog.Builder
import androidx.appcompat.app.AppCompatActivity
import zac4j.com.webapp.R.id
import zac4j.com.webapp.R.layout

/**
 * Activity for web app.
 * Created by Zaccc on 2017/7/20.
 */
class WebAppActivity : AppCompatActivity() {
  private lateinit var mBrowser: WebView

  @SuppressLint("SetJavaScriptEnabled")
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(layout.activity_js_alert)
    mBrowser = findViewById<View>(id.main_webview) as WebView
    mBrowser.settings.javaScriptEnabled = true
    mBrowser.webChromeClient = MyWebChromeClient()
    mBrowser.loadUrl("file:///android_asset/prompt.html")
  }

  private inner class MyWebChromeClient : WebChromeClient() {
    override fun onJsAlert(
      view: WebView,
      url: String,
      message: String,
      result: JsResult
    ): Boolean {
      Builder(this@WebAppActivity)
          .setTitle("WebApp Alert")
          .setMessage(message)
          .setPositiveButton(string.ok) { dialog, which -> result.confirm() }
          .setCancelable(false)
          .create()
          .show()
      return true
    }

    override fun onJsConfirm(
      view: WebView,
      url: String,
      message: String,
      result: JsResult
    ): Boolean {
      Builder(this@WebAppActivity)
          .setTitle("WebApp Confirm")
          .setMessage(message)
          .setPositiveButton(string.ok) { dialog, which -> result.confirm() }
          .setCancelable(false)
          .create()
          .show()
      return true
    }

    override fun onJsPrompt(
      view: WebView,
      url: String,
      message: String,
      defaultValue: String,
      result: JsPromptResult
    ): Boolean {
      Builder(this@WebAppActivity)
          .setTitle("WebApp Prompt")
          .setMessage(message)
          .setPositiveButton(string.ok) { dialog, which -> result.confirm() }
          .setCancelable(false)
          .create()
          .show()
      return true
    }
  }
}